### PR TITLE
EN-6013: heartbeat update for peer list

### PR DESCRIPTION
**Pull request by bogdan-rosianu**

### CODE REVIEW

File: node/heartbeat/heartbeatMessageInfo.go [hunk 1]
```diff
@@ -192,6 +195,5 @@ func (hbmi *heartbeatMessageInfo) GetIsActive() bool {
 func (hbmi *heartbeatMessageInfo) GetIsValidator() bool {
 	hbmi.updateMutex.Lock()
 	defer hbmi.updateMutex.Unlock()
-	isValidator := hbmi.isValidator
-	return isValidator
+	return hbmi.peerType == string(core.EligibleList)
```
Comment (AdoAdoAdo) [inline node/heartbeat/heartbeatMessageInfo.go]: it is a validator if it is in waiting list as well

File: node/heartbeat/sender.go [hunk 2]
```diff
@@ -104,3 +119,18 @@ func (s *Sender) SendHeartbeat() error {
...
+
+func (s *Sender) updateMetrics(hb *Heartbeat) {
+	result := s.computePeerList(hb.Pubkey)
+	s.statusHandler.SetStringValue(core.MetricPeerType, result)
+}
+
+func (s *Sender) computePeerList(pubkey []byte) string {
+	peerType, _, err := s.peerTypeProvider.ComputeForPubKey(pubkey)
+	if err != nil {
+		log.Warn("monitor: compute peer type", "error", err)
```
Comment (AdoAdoAdo) [inline node/heartbeat/sender.go]: warn message is incorrect, this is not in monitor, this is in sender

File: node/heartbeat/heartbeatMessageInfo.go (hunk 1 above)
Comment (bogdan-rosianu) [reply node/heartbeat/heartbeatMessageInfo.go]: fixed

File: node/heartbeat/sender.go (hunk 2 above)
Comment (bogdan-rosianu) [reply node/heartbeat/sender.go]: done

File: node/heartbeat/heartbeatMessageInfo.go [hunk 3]
```diff
@@ -192,6 +195,5 @@ func (hbmi *heartbeatMessageInfo) GetIsActive() bool {
 func (hbmi *heartbeatMessageInfo) GetIsValidator() bool {
 	hbmi.updateMutex.Lock()
 	defer hbmi.updateMutex.Unlock()
-	isValidator := hbmi.isValidator
-	return isValidator
+	return hbmi.peerType == string(core.EligibleList) || hbmi.peerType == string(core.WaitingList)
```
Comment (raduchis) [inline node/heartbeat/heartbeatMessageInfo.go]: can you please add a unit test for GetIsValidator? to see that the isValidator works with peerType eligible, waiting and another dummy list

File: node/heartbeat/monitor.go [hunk 4]
```diff
@@ -160,7 +167,9 @@ func (m *Monitor) loadRestOfPubKeysFromStorage() error {
 	for _, peer := range peersSlice {
 		pubKey := string(peer)
 		_, ok := m.heartbeatMessages[pubKey]
+		log.Info("loaded heartbeat for pubkey from storage", "pubkey", peer)
```
Comment (raduchis) [inline node/heartbeat/monitor.go]: is the info needed here? Could it be just on debug?

Comment (bogdan-rosianu) [reply node/heartbeat/monitor.go]: sorry, just used it for debugging. will remove it

File: node/heartbeat/heartbeatMessageInfo.go [hunk 5]
```diff
@@ -145,6 +146,7 @@ func (hbmi *heartbeatMessageInfo) HeartbeatReceived(
 	receivedshardID uint32,
```
[the remark this answers is not in the document]
Comment (bogdan-rosianu) [reply node/heartbeat/heartbeatMessageInfo.go]: done

File: node/heartbeat/heartbeatMessageInfo.go (hunk 3 above)
Comment (bogdan-rosianu) [reply node/heartbeat/heartbeatMessageInfo.go]: added 3 unit tests

File: node/heartbeat/monitor.go (hunk 4 above)
Comment (raduchis) [reply node/heartbeat/monitor.go]: ok

File: node/heartbeat/heartbeatMessageInfo.go (hunk 3 above)
Comment (raduchis) [reply node/heartbeat/heartbeatMessageInfo.go]: ok

**Pull request merged**